### PR TITLE
Build updates for Cantera 3.0 features

### LINKED ISSUE #43: Link conda packages to libhdf5
Cantera 3.0 adds "native" support for working with HDF5 files using libhdf5. Supporting this feature with the conda packages requires linking to a specific version of hdf5, and we need to decide how to handle this.

Some possibilities:
- Link against the latest packaged version of `libhdf5` and leave it at that
- Provide packages linking to different versions of `libhdf5`, with the hdf5 version included in the version tag
- Provide packages with and without `libhdf5` dependencies, with the version tag specified so that installing with HDF5 is preferred

I think the last option is probably the best way to avoid creating dependency problems that prevent users from installing specific Cantera versions without downgrading other packages (for example, when users who also want to have `h5py` installed install a version of that that link to a newer version of HDF5 than is compatible with a given Cantera release).

### PR DESCRIPTION
**Pull request by speth**
- Build with support for "native" HDF5. If we just want to build with support for the latest HDF5 version, this is all that's needed.
- Install the plugin module needed for calling `ExtensibleRate` implementations from programs linking to the Cantera shared library as part of the corresponding version of the Python module.
- Add builds for Python 3.11
- Fix some miscellaneous build issues.

Fixes #43 

See CI status here: https://github.com/speth/conda-recipes/actions/workflows/main.yml

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -151,6 +154,8 @@ jobs:
   build-matlab:
     name: Matlab toolbox for ${{ matrix.os }}
     runs-on: ${{ matrix.os }}
+    # Run this job even if post-pending-status was skipped
+    if: always()
```
Comment (bryanwweber) [inline .github/workflows/main.yml:158]: These should probably be `success() || cancelled()` so they don't run if the previous step actually failed.

Comment (speth) [reply .github/workflows/main.yml:158]: Is skipped the same as cancelled?

Even so, if the "post-pending-status" job fails, I think I'd still rather go ahead with the actual package build.

**Review by ischoegl — APPROVED** [review]
This looks good to me, with the caveat that I'm not super-experienced with recipes. I'll defer to @bryanwweber, but am approving anyways.

File: .github/workflows/main.yml (hunk 1 above)
Comment (bryanwweber) [reply .github/workflows/main.yml:167]: Yeah, I'm pretty sure that's the case. Anyways, your reasoning seems fine, it's good with me to keep it this way.

Comment (bryanwweber) [reply .github/workflows/main.yml:167]: Yeah, I'm pretty sure that's the case. Anyways, your reasoning seems fine, it's good with me to keep it this way.

**Review by bryanwweber — APPROVED** [review]

**Pull request merged**